treeherder/model/error_summary.py

```diff
@@ -32,15 +33,20 @@ def get_error_summary(job):
 
     # don't cache or do anything if we have no text log errors to get
     # results for
-    errors = TextLogError.objects.filter(step__job=job)
+    errors = TextLogError.objects.filter(job=job)
     if not errors:
         return []
 
     # cache terms generated from error line to save excessive querying
     term_cache = {}
 
     error_summary = [bug_suggestions_line(err, term_cache) for err in errors]
-    cache.set(cache_key, error_summary, BUG_SUGGESTION_CACHE_TIMEOUT)
+
+    try:
+        cache.set(cache_key, error_summary, BUG_SUGGESTION_CACHE_TIMEOUT)
+    except Exception as e:
+        newrelic.agent.record_custom_event('error caching error_summary for job', job.id)
+        logger.error('error caching error_summary for job %s: %s', job.id, e)
```

Comment:
If you use `exc_info` you can keep track of the Traceback as well in the output.

```python
import logging
logging.basicConfig()
logging.getLogger().setLevel(logging.INFO)

job = {'id': 1}
try:
    raise Exception('My message')
except Exception as e:
    logging.warning('Foo2 {} {}'.format(job['id'], e), exc_info=True)
    logging.warning('Foo2 %s %s', job['id'], e, exc_info=True)

logging.info('We survivided the try block!')
```

```shell
WARNING:root:Foo2 1 My message
Traceback (most recent call last):
  File "foo.py", line 7, in <module>
    raise Exception('My message')
Exception: My message
WARNING:root:Foo2 1 My message
Traceback (most recent call last):
  File "foo.py", line 7, in <module>
    raise Exception('My message')
Exception: My message
INFO:root:We survivided the try block!
```

Reply:
oh cool, I didn't know about that. I'll add it in.